### PR TITLE
Extra 

### PR DESCRIPTION
**Pull request by JRNeal2020**

### CODE REVIEW

**Review by JesseTNRoberts — COMMENTED** [review]
Any lights? Any sounds? That seems very open. Are there no constraints at all?

**Review by JesseTNRoberts — COMMENTED** [review]
So lights that are invisible are sufficient? How about sounds that are not audible?

File: Documentation/Signoffs/Extra.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+- The goal of this subsystem is to adhere to the rules as provided by the customer, Devcom, and to add lights and sounds to the design which will add to the point total as provided by Devcom. The system needs a pause switch to deactivate it between rounds. The system will also include lights and sounds before launching a projectile to add to the decoration portion of the design.
+
+## **Constraints:**
+
+| **Number:** | **Constraint:** | **Origin:** | 
+| --- | --- | --- |
+| 1. |  The interceptor shall have a switch that sets the system into a pause state that will keep the interceptor from firing. | Rulebook |
+| 2. | The voltage switched by the pause switch shall be 5V. | System Constraint|
+| 3. | The interceptor must have lights that are bright enough to be seen by the judges. These lights must be powered by a 5V source from the processor block and must emit a light that falls within the visual light spectrum of 380 to 720 nanometers [1]. | Rulebook / System |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Extra.md]: How bright do they need to be?

File: Documentation/Signoffs/Extra.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+| 4. | The interceptor must make sounds before firing. The sounds will be powered by a voltage of 5V and the sounds will need to fall within the range of human hearing which is 20 to 20,000Hz or 0 - 120db [2]. The sound that is played should not damage anyone's ears therefore it should not be over 85db for a long period of time [2]. | Rulebook / Safety |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Extra.md]: db and hz are not interchangeable and do not convey similar information. This constraint needs to be reconsidered and background reading on what is conveyed by db and hz is needed.

File: Documentation/Signoffs/Extra.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+   
+5. To add to the total points for the competition, the interceptor must make sounds before firing. This constraint can be found in the official rulebook for the competition. The piezo buzzer must be able to work with an output of the Arduino Uno R3 which will be 5V. It must also fall within the human hearing range of frequency which is 20 to approximately 20,000Hz. The piezo buzzer that was chosen operates at a frequency of 3600Hz to 4600Hz which will fall within the frequency range that we can hear. When it comes to loudness, humans can tolerate a range of 0 to 120db and the sound must not exceed 85db for long periods which will keep the noise at a save level [2]. The buzzer that has been chosen operates from 3 to 16 VDC and operates at a minimum of 70db at 12V therefore at the voltage that is being used, the sounds will be at a safe level for the small periods of time that they will be used.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Extra.md]: I was unaware that 0db would be considered audible.

**Review by JesseTNRoberts — COMMENTED** [review]
You analysis does not address the constraints.

File: Documentation/Signoffs/Extra.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+- The goal of this subsystem is to adhere to the rules as provided by the customer, Devcom, and to add lights and sounds to the design which will add to the point total as provided by Devcom. The system needs a pause switch to deactivate it between rounds. The system will also include lights and sounds before launching a projectile to add to the decoration portion of the design.
+
+## **Constraints:**
+
+| **Number:** | **Constraint:** | **Origin:** | 
+| --- | --- | --- |
+| 1. |  The interceptor shall have a switch that sets the system into a pause state that will keep the interceptor from firing. | Rulebook |
+| 2. | The voltage switched by the pause switch shall be 5V. | System Constraint|
+| 3. | The interceptor must have lights that are bright enough to be seen by the judges approximately six and a half feet away, which is the longest length of the gameboard. These lights must also be powered by a 5V source from the processor block and must emit a light that falls within the visual light spectrum of 380 to 720 nanometers [1]. | Rulebook / System |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Extra.md]: Why are the lights required to be 5volts?

Comment (JesseTNRoberts) [reply Documentation/Signoffs/Extra.md]: So, any light, no matter how dim, in that spectrum is sufficient?

File: Documentation/Signoffs/Extra.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+| 4. | The interceptor must make sounds before firing. The sounds will be powered by a voltage of 5V and the sounds will need to fall within the range of frequency humans can hear which is 20 to 20,000Hz [2]. The volume, or loudness, must fall within a range of 0, the threshold, to a maximum of 120dB and must not exceed 85dB for more than eight hours or it may cause hearing damage [2]. The buzzer should be louder than average room noise which is 30-50 dB [2]. | Rulebook / Safety |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Extra.md]: What does 0, the threshold, mean? 

This constraint still doesn't actually constrain the system along the important lines and instead tries to require that the buzzer operate on 5 volts. That's not a legitimate constraint. Remove it. In its place, give the constraints for this system. Below, provide the clear reasoning behind the constraint.

File: Documentation/Signoffs/Extra.md [hunk 6]
```diff
@@ -12,44 +12,41 @@ Figure 1: Extra subsystem and the pause switch can be seen in this figure.
...
+| 3. | The interceptor must have lights that are bright enough to be seen by the judges approximately six and a half feet away, which is the longest length of the gameboard. Therefore the lights must have a candela rating greater than that of a fire alarm which is 15 cd [1]. These lights must also emit a light that falls within the visual light spectrum of 380 to 720 nanometers [2]. | Rulebook / System |
+| 4. | The interceptor must make sounds before firing. The sounds will need to fall within the range of frequency humans can hear which is 20 to 20,000Hz [3]. The volume, or loudness, must fall within a range of 30-50 dB, which is the average noise of a room [3], to a maximum of 120dB and must not exceed 85dB for more than eight hours or it may cause hearing damage [3]. | Rulebook / Safety |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/Extra.md]: Why this range of db? This suggests that you wouldn't hear the siren for the room noise. Think about the logic you applied to the lighting and apply similar logic here.

**Comment by micahr234** [comment]
In the opening description explain exactly how the lights and button will work. Do more than "to make Devcom happy".

Can you resistors handle the power they will have to dissipate? For red? For green?

Normally microprocessors have an internal pull-down resistor, that way you do not have to have a double pole switch.

How about a PCB board? How are you going to connect all this together?

**Comment by micahr234** [comment]
I believe your power calculations are wrong.

Your schematic does not qualify as a PCB layout. Use KiCAD to make a PCB layout.

File: Documentation/Signoffs/Extra.md
Comment (micahr234) [inline Documentation/Signoffs/Extra.md:1]: I believe your power calculations are wrong for the power dissipated by resistor. Red 2.5V x 50mA = 125mW. You find the green resistor power.

Use KiCAD to make a PCB layout.

**Review by micahr234 — APPROVED** [review]

**Pull request merged**